tests/test_core.py

```diff
@@ -6417,23 +6417,25 @@ def test_large_exported_response(self):
     assert '_exported_func_from_response_file_1' in open('src.cpp.o.js').read()
 
   @sync
-  @no_wasm_backend('no jsCall function pointers are created for wasm backend')
   def test_add_function(self):
     Settings.INVOKE_RUN = 0
     Settings.RESERVED_FUNCTION_POINTERS = 1
 
     test_path = path_from_root('tests', 'interop')
     src, expected = (os.path.join(test_path, s) for s in ('test_add_function.cpp', 'test_add_function.out'))
 
-    post_js = os.path.join(test_path, 'test_add_function_post.js')
+    if Settings.WASM_BACKEND:
+      post_js = os.path.join(test_path, 'test_add_function_post.js')
```

Comment:
Should probably just use `test_add_function.js`, and leave the signature there. A desirable property to have would be if users can build for either asm.js or wasm with the same codebase, which means they should be able to use `addFunction` with the signature in asm.js without anything breaking, and we should probably test that.

Alternatively: keep both .js files, always test with the signature, test without the signature if not WASM_BACKEND (because no-sig in asm.js shouldn't break either)

Reply:
I guess just testing `test_add_function.js` would be ok..? (I had forgotten about that JS functions don't mind being called with different number of arguments, and that was the reason why I split the file)

The current `test_add_function` has 5 tests in total, some of them only test wasm path and the others asm.js path, so I'm wondering if adding another dimension to each of them makes the code overly complicated.